File: src/Backend/Services/Catalog/Domains.Article/Operations/List/Get/ArticleDomainListGetOperationRequest.cs
// Copyright (c) 2023 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Crib2023.Backend.Services.Catalog.Domains.Article.Operations.List.Get;

/// <summary>
/// Запрос операции получения списка в домене.
/// </summary>
public class ArticleDomainListGetOperationRequest : IRequest<ArticleDomainListGetOperationResponse>
{
    #region Properties

    /// <summary>
    /// Входные данные.
    /// </summary>
    public ArticleListGetOperationInput Input { get; }

    /// <summary>
    /// Код операции.
    /// </summary>
    public string OperationCode { get; }

    #endregion Properties

    #region Constructors

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="input">Входные данные.</param>
    /// <param name="operationCode">Код операции.</param>
    public ArticleDomainListGetOperationRequest(ArticleListGetOperationInput input, string operationCode = "")
    {
        Input = input;
        OperationCode = operationCode;
    }

    #endregion Constructors
}

File: src/Frontend/React/TypeScript/App/src/domains/Article/Operations/List/Get/Request/ArticleDomainListGetOperationRequest.ts
import {
  type ApiRequestWithInput,
  type ArticleDomainListGetOperationInput,
} from '../../../../../../all';

export interface ArticleDomainListGetOperationRequest
  extends ApiRequestWithInput<ArticleDomainListGetOperationInput> {}

export function createArticleDomainListGetOperationRequest (
  input: ArticleDomainListGetOperationInput,
  operationCode: string = ''
): ArticleDomainListGetOperationRequest {
  return {
    operationName: '@@ArticleDomainListGet',
    operationCode,
    input
  };
}
